### ISSUE TITLE
Query using the LN() function does not terminate

### ISSUE THREAD

**Issue by mrigger** [cwida/duckdb#518]
Consider the following statements:
```sql
CREATE TABLE t0(c0 INT);
CREATE TABLE t1(c0 INT);
INSERT INTO t0(c0) VALUES (0);
INSERT INTO t1(c0) VALUES (0), (0), (1), (-1);
SELECT * FROM t0, t1 WHERE LN(t1.c0) < t0.c0; -- does not terminate
```
Unexpectedly, the `SELECT` does not terminate.

I found this bug based on the latest master commit (9795d18b6a6e250a4076fa448f248287e76f5693).

**Comment by Mytherin** [comment]
Very interesting! I wonder what causes this :)

**Comment by Mytherin** [comment]
Fixed in 230785f2e6dceac410cfd894db214d21e356b127. The issue was that LN(-1) or LN(0) produced a NaN value, which resulted in an infinite loop in the quick sort step of the merge join. The reason for that is that we had a loop in the form of:
```cpp
while(i < j) {
    while (data[i] < data[pivot]) {
         i++;
    }
    while (data[pivot] < data[j]) {
        j--;
    }
    if (i < j) {
       swap(data[i], data[j]);
    }
}
```

However, since NaN always compares false with any other value, this loop became infinite as the conditions to trigger either `i++` or `j--` were never triggered.

I fixed this case by filtering out NaN values prior to the sort, similar to how NULL values were handled, and also changed the LN, LOG2, LOG10 and SQRT operators to return in a NULL if given invalid input, rather than potentially producing NaN or infinite.
